### ISSUE TITLE
AttributeError: 'ForkAwareLocal' object has no attribute 'connection'

### ISSUE THREAD

**Issue by npip99** [itaicaspi/remote_plot#1]
Hello!

This seems like a super-useful repo, X11 forwarding of matplotlib sucks!

However, there seems to be an issue when "end-of-file" has been reached

```
ubuntu@ip-123-45-6-7:~/my_dir$ python3 graph_book.py 
Serving HTTP on 0.0.0.0 port 8000 (http://0.0.0.0:8000/) ...
----------------------------------------
Exception occurred during processing of request from ('127.0.0.1', 60430)
Traceback (most recent call last):
  File "/usr/lib/python3.10/multiprocessing/managers.py", line 810, in _callmethod
    conn = self._tls.connection
AttributeError: 'ForkAwareLocal' object has no attribute 'connection'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.10/socketserver.py", line 683, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/lib/python3.10/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/lib/python3.10/socketserver.py", line 747, in __init__
    self.handle()
  File "/usr/lib/python3.10/http/server.py", line 432, in handle
    self.handle_one_request()
  File "/usr/lib/python3.10/http/server.py", line 420, in handle_one_request
    method()
  File "/home/ubuntu/.local/lib/python3.10/site-packages/remote_plot/server.py", line 96, in do_GET
    token = self.server.SHARED['token']
  File "<string>", line 2, in __getitem__
  File "/usr/lib/python3.10/multiprocessing/managers.py", line 814, in _callmethod
    self._connect()
  File "/usr/lib/python3.10/multiprocessing/managers.py", line 801, in _connect
    conn = self._Client(self._token.address, authkey=self._authkey)
  File "/usr/lib/python3.10/multiprocessing/connection.py", line 502, in Client
    c = SocketClient(address)
  File "/usr/lib/python3.10/multiprocessing/connection.py", line 630, in SocketClient
    s.connect(address)
FileNotFoundError: [Errno 2] No such file or directory
----------------------------------------
----------------------------------------
Exception occurred during processing of request from ('127.0.0.1', 60432)
Traceback (most recent call last):
  File "/usr/lib/python3.10/multiprocessing/managers.py", line 810, in _callmethod
    conn = self._tls.connection
AttributeError: 'ForkAwareLocal' object has no attribute 'connection'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.10/socketserver.py", line 683, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/lib/python3.10/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/lib/python3.10/socketserver.py", line 747, in __init__
    self.handle()
  File "/usr/lib/python3.10/http/server.py", line 432, in handle
    self.handle_one_request()
  File "/usr/lib/python3.10/http/server.py", line 420, in handle_one_request
    method()
  File "/home/ubuntu/.local/lib/python3.10/site-packages/remote_plot/server.py", line 96, in do_GET
    token = self.server.SHARED['token']
  File "<string>", line 2, in __getitem__
  File "/usr/lib/python3.10/multiprocessing/managers.py", line 814, in _callmethod
    self._connect()
  File "/usr/lib/python3.10/multiprocessing/managers.py", line 801, in _connect
    conn = self._Client(self._token.address, authkey=self._authkey)
  File "/usr/lib/python3.10/multiprocessing/connection.py", line 502, in Client
    c = SocketClient(address)
  File "/usr/lib/python3.10/multiprocessing/connection.py", line 630, in SocketClient
    s.connect(address)
FileNotFoundError: [Errno 2] No such file or directory
----------------------------------------
```

The same issues happen if I run `plt.plot()` in the python shell, and then hit "Cmd+D" for the EOF character. It's not a big deal, just CTRL+C exits, but just thought I'd post the error here.

**Comment by npip99** [comment]
[Separate question, would it ever be possible to get overlapping plots to work? It's often useful to be able to call `.plot` multiple times, in combination with `.legend` and `.show`, to see multiple different data series simultaneously.]

**Comment by itaicaspi** [comment]
Hey @npip99 thanks for bringing this up.
I was able to reproduce the issue you are facing and will look into it.

Regarding overlaying plots, it is definitely possible. You can currently add the flag `clear_figure=False` to any call in order to overlay it and prevent it from clearing the previous plot.
for example,
```
from remote_plot import plt

plt.bar([1, 2, 3], [4, 5, 6])
plt.bar([1, 2, 4], [7, 8, 9], clear_figure=False)
```

But my intention is for this library to match matplotlibs api as close as possible so this is a temporary solution.
Also `.legend` should definitely be possible, and I will add it to the TODO list.

**Comment by itaicaspi** [comment]
Fixed the issue you were facing in 741b15891c7402b3017c3ce38ec31b7d3d33e25c
It happened due to the server not being closed when the script ends.

Also updated the API, check the [examples directory](https://github.com/itaicaspi/remote_plot/tree/main/examples) for a few usage examples, including using `.legend` and `.show`
